### PR TITLE
refactor(slider): upgrade rc-slider, split range, tooltip from slider. close #777

### LINKED ISSUE #777: slider refactor
## current problem
- too big for common usage: http://facebook.github.io/react-native/docs/slider.html

## solution
- split tooltip/range functionality from simple usage

```
import {Slider, Ranger, SliderTooltipPlugin} from 'antd-mobile'

<Slider plugin={[TooltipPlugin]}>

<Ranger plugin={[TooltipPlugin]}>
```

### PR DESCRIPTION
**Pull request by paranoidjk**
Ref: #777 

First of all, thanks for your contribution! :-)

Please makes sure these boxes are checked before submitting your PR, thank you!

* [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

**Comment by paranoidjk** [comment]
ping @warmhug @silentcloud #777 break change, review下

**Comment by benjycui** [comment]
Done 了 ping 下我

**Comment by paranoidjk** [comment]
ping @benjycui @yiminghe @warmhug @silentcloud done， 这个ok了

File: components/slider/demo/basic.md [hunk 1]
```diff
@@ -9,24 +9,48 @@ title: 基本
...
         </WingBlank>
         <WhiteSpace size="lg" />
+        <WingBlank size="lg">
+          <p className="title">Range, 基础使用</p>
+          <Range min={0} max={20} defaultValue={[3, 10]} onChange={this.log('change')} onAfterChange={this.log('afterChange')} />
+        </WingBlank>
+        <WhiteSpace size="lg" />
+        <WingBlank size="lg">
+          <p className="title">Range, 注入 SliderTooltipPlugin 插件</p>
+          <Range min={0} max={20} defaultValue={[3, 10]} plugin={[SliderTooltipPlugin]} />
```
Comment (benjycui) [inline components/slider/demo/basic.md]: 其实我觉得现阶段没必要加上数组啊。。。

以后真的有多个的时候，再支持吧。

**Comment by benjycui** [comment]
@yiminghe 确认要以 plugin 的形式存在么。。

File: components/slider/demo/basic.md (hunk 1 above)
Comment (paranoidjk) [inline components/slider/demo/basic.md]: 我内部是支持了的，多个 HOC 层层包装而已，不过对外 API 会变复杂，看起来不友好

Comment (yiminghe) [inline components/slider/demo/basic.md]: plugins 吧  https://www.sencha.com/blog/using-plugins-and-mixins-in-your-sencha-apps/

Comment (yiminghe) [inline components/slider/demo/basic.md]: 除了 plugin 还有什么好方式么？

Comment (benjycui) [inline components/slider/demo/basic.md]: See: http://react-component.github.io/slider/examples/handle.html

直接用 HOC 吧，毕竟 HOC 是社区普遍接受的概念，而 plugin 则是私有的概念。

Comment (paranoidjk) [inline components/slider/demo/basic.md]: ## Plugin
```
import { Range, SliderTooltipPlugin } from 'antd-mobile';

<Range min={0} max={20} defaultValue={[3, 10]} plugin={[SliderTooltipPlugin]} />
```

## Hoc
```
import { Range, CreateSliderWithTooltip } from 'antd-mobile';

const RangeWithTooltip = CreateSliderWithTooltip(Range);
<RangeWithTooltip min={0} max={20} defaultValue={[3, 10]}  />
```

@ant-design/ant-design-team 
这个 API 需要定下来，下午准备 release `1.0` 了

Comment (yiminghe) [inline components/slider/demo/basic.md]: 都行，就按作者 @benjycui 定的吧

Comment (yesmeck) [inline components/slider/demo/basic.md]: HOC +1

**Pull request merged**